addons/website_sale/models/sale_order_line.py

```diff
@@ -90,14 +90,18 @@ def _is_reorder_allowed(self):
             and self._show_in_cart()
         )
 
-    def _get_cart_display_price(self):
+    def _get_cart_display_price(self, qty=None):
         self.ensure_one()
         price_type = (
             'price_subtotal'
             if self.order_id.website_id.show_line_subtotals_tax_selection == 'tax_excluded'
             else 'price_total'
         )
-        return sum(self._get_lines_with_price().mapped(price_type))
+        display_price = sum(self._get_lines_with_price().mapped(price_type))
+        if qty and self.product_qty:
+            # display the price for [qty] products only
+            display_price *= qty / self.product_qty
+        return display_price
```

Comment:
Although I understand the benefits of adding `qty` here, I think it's not the right place to do it. Indeed, a `SaleOrderLine` already has a quantity set on it. IMO, there is no other case where you would want to compute the price partially (by changing the qty). 

Either this method should use the quantity of the line or just return the unit price. As we are targeting a stable version, I would advise the latter, so keep it as it is right now. You can add a new method in a utils file or directly in the controller if you don't want to make the computation of the total price for the added products in the route.